pwiz_tools/Skyline/SkylineTester/SkylineTesterWindow.cs

```diff
@@ -1,1796 +1,1866 @@
+                formsLanguageDiff.Enabled = false;
+                diffButton.Enabled = false;
+            }
+            else
+            {
+                PopulateFormsLanguageDiff(formsLanguage.SelectedItem.ToString());
+            }
+        }
+
+        private void formsLanguageDiff_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            if (formsLanguageDiff.SelectedItem.ToString() == "")
+                diffButton.Enabled = false;
+            else
+                diffButton.Enabled = true;
+        }
+
+        private string[] GetChangedFileList()
+        {
+            var skylineDir = Path.GetFullPath(Path.Combine(ExeDir, @"..\..\.."));
+            var procStartInfo = new ProcessStartInfo("cmd", "/c " + skylineDir + " & git diff --name-only");
+
+            procStartInfo.RedirectStandardOutput = true;
+            procStartInfo.UseShellExecute = false;
+            procStartInfo.CreateNoWindow = true;
+            var cmd = new Process();
+            cmd.StartInfo = procStartInfo;
+            cmd.Start();
+
+            var changedFiles = cmd.StandardOutput.ReadToEnd();
+            return changedFiles.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
+        }
+
+        private static string GetFileName(string file)
+        {
+            var fileName = Path.GetFileName(file);
+            return fileName.Substring(0, fileName.IndexOf(".", StringComparison.Ordinal));
+        }
+
+        private void showChangedFiles_CheckedChanged(object sender, EventArgs e)
+        {
+            var changedFilesArray = GetChangedFileList();
+
+            if (showChangedFiles.Checked)
+            {
+                foreach (DataGridViewRow row in formsGrid.Rows)
+                {
+                    var formNoExt = "";
+                    if (row.Cells[0].Value.ToString().Contains("."))
+                        formNoExt = row.Cells[0].Value.ToString().Substring(0,
+                            row.Cells[0].Value.ToString().IndexOf(".", StringComparison.Ordinal));
+                    else
+                        formNoExt = row.Cells[0].Value.ToString();
+
+                    foreach (var file in changedFilesArray)
+                    {
+                        if (!Equals(file, ""))
+                        {
+                            var fileName = GetFileName(file);
+                            if (Equals(formNoExt, fileName))
+                            {
+                                row.Visible = true;
+                                break;
+                            }
+                            else
+                            {
+                                row.Visible = false;
+                            }
+                        }
+                    }
+                }
+                formsGrid.Update();
+            }
+            else
+            {
+                foreach (DataGridViewRow row in formsGrid.Rows)
```

Comment:
And you could make it just one loop with the single line:

row.Visible = !showChangedFiles.Checked || changedFileSet.Contains(formNoExt);